Integrations/MicrosoftGraphGroups/MicrosoftGraphGroups.py

```diff
@@ -0,0 +1,441 @@
+        if dbot_response.status_code not in {200, 201}:
+            msg = 'Error in authentication. Try checking the credentials you entered.'
+            try:
+                demisto.info('Authentication failure from server: {} {} {}'.format(
+                    dbot_response.status_code, dbot_response.reason, dbot_response.text))
+                err_response = dbot_response.json()
+                server_msg = err_response.get('message')
+                if not server_msg:
+                    title = err_response.get('title')
+                    detail = err_response.get('detail')
+                    if title:
+                        server_msg = f'{title}. {detail}'
+                if server_msg:
+                    msg += ' Server message: {}'.format(server_msg)
+            except Exception as ex:
+                demisto.error('Failed parsing error response - Exception: {}'.format(ex))
+            raise Exception(msg)
+        try:
+            gcloud_function_exec_id = dbot_response.headers.get('Function-Execution-Id')
+            demisto.info(f'Google Cloud Function Execution ID: {gcloud_function_exec_id}')
+            parsed_response = dbot_response.json()
+        except ValueError:
+            raise Exception(
+                'There was a problem in retrieving an updated access token.\n'
+                'The response from the Demistobot server did not contain the expected content.'
+            )
+        access_token = parsed_response.get('access_token')
+        expires_in = parsed_response.get('expires_in', 3595)
+        time_now = epoch_seconds()
+        time_buffer = 5  # seconds by which to shorten the validity period
+        if expires_in - time_buffer > 0:
+            # err on the side of caution with a slightly shorter access token validity period
+            expires_in = expires_in - time_buffer
+
+        demisto.setIntegrationContext({
+            'access_token': access_token,
+            'valid_until': time_now + expires_in
+        })
+        return access_token
+
+    def http_request(self, method: str, url_suffix: str = None, params: Dict = None, body: Optional[str] = None,
+                     next_link: str = None):
+        """
+        Generic request to Microsoft Graph
+        """
+        token = self.get_access_token()
+        if next_link:
+            url = next_link
+        else:
+            url = f'{self.base_url}{url_suffix}'
+
+        response = requests.request(
+            method,
+            url,
+            headers={
+                'Authorization': 'Bearer ' + token,
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            },
+            params=params,
+            data=body,
+            verify=self.use_ssl,
+        )
+        try:
+            data = response.json() if response.text else {}
+            if not response.ok:
+                return_error(f'API call to MS Graph failed [{response.status_code}]'
```

Comment:
you should raise formatted exception instead of return error. this should be handled in the main func.

Reply:
@bakatzir seems you missed that comment